src/main/java/fi/aalto/cs/apluscourses/ui/exercise/SubmissionDialog.java

```diff
@@ -81,12 +81,12 @@ private void createUIComponents() {
     filenamesHtml.append("</ul></body></html>");
     filenames = new JLabel(filenamesHtml.toString());
 
-    submissionCount = new JLabel(String.format("You are about to make submission %d out of %d.",
-        viewModel.getCurrentSubmissionNumber(), viewModel.getMaxNumberOfSubmissions()));
+    StringBuilder submissionCountText = new StringBuilder("You are about to make submission "
+        + viewModel.getCurrentSubmissionNumber());
+    if (viewModel.getMaxNumberOfSubmissions() != 0) {
+      submissionCountText.append(" out of " + viewModel.getMaxNumberOfSubmissions());
+    }
+    submissionCount = new JLabel(submissionCountText.toString() + ".");
```

Comment:
Nit: This logic could be in the view model (and unit-tested there).

Reply:
Also include `"."` at the end in the builder. :D

Reply:
You are correct on both points :smile: will do